File: examples/gimbal_move_angle.cpp
#include "stdio.h"
#include <pthread.h>
#include <cstdlib>
#include <string>
using namespace std;

#include"payloadSdkInterface.h"

#if (CONTROL_METHOD == CONTROL_UART)
T_ConnInfo s_conn = {
    CONTROL_UART,
    payload_uart_port,
    payload_uart_baud
};
#else
T_ConnInfo s_conn = {
    CONTROL_UDP,
    udp_ip_target,
    udp_port_target
};
#endif

PayloadSdkInterface* my_payload = nullptr;

pthread_t thrd_recv;
pthread_t thrd_gstreamer;

bool gstreamer_start();
void gstreamer_terminate();
void *start_loop_thread(void *threadid);


bool all_threads_init();
void quit_handler(int sig);


int main(int argc, char *argv[]){
	printf("Starting Set gimbal mode example...\n");
	signal(SIGINT,quit_handler);

	// create payloadsdk object
	my_payload = new PayloadSdkInterface(s_conn);

	// init payload
	my_payload->sdkInitConnection();
	printf("Waiting for payload signal! \n");

	my_payload->checkPayloadConnection();
	usleep(100000);

	printf("Set gimbal RC mode \n");
	my_payload->setPayloadCameraParam(PAYLOAD_CAMERA_RC_MODE, PAYLOAD_CAMERA_RC_MODE_STANDARD, PARAM_TYPE_UINT32);
	usleep(100000);

	printf("Move gimbal yaw to 90 deg, delay in 5secs \n");
	my_payload->setGimbalSpeed(0, 0 , 90, INPUT_ANGLE);
	usleep(5000000);

	printf("Move gimbal yaw to -90 deg, delay in 5secs \n");
	my_payload->setGimbalSpeed(-0, -0 , -90, INPUT_ANGLE);
	usleep(5000000);

	printf("Move gimbal yaw to 0 deg, delay in 5secs \n");
	my_payload->setGimbalSpeed(0, 0 , 0, INPUT_ANGLE);
	usleep(500000);

	// close payload interface
	try {
		my_payload->sdkQuit();
	}
	catch (int error){}

	return 0;
}

void quit_handler( int sig ){
    printf("\n");
    printf("TERMINATING AT USER REQUEST \n");
    printf("\n");

    // close payload interface
    try {
        my_payload->sdkQuit();
    }
    catch (int error){}

    // end program here
    exit(0);
}

File: gimbal_move_angle.py
import time
import signal
import sys
import os
from libs.payload_sdk import PayloadSdkInterface, input_mode_t, param_type
from libs.payload_define import *

my_payload = None

# Signal handler for quitting
def quit_handler(sig, frame):
    global my_payload
    print("\nTERMINATING AT USER REQUEST")

    # Close payload interface
    if my_payload:
        try:
            my_payload.sdkQuit()
        except Exception as e:
            print(f"Error while quitting payload: {e}")
    # End program    
    sys.exit(0)

def main():
    global my_payload

    print("Starting Set gimbal mode example...\n")
    signal.signal(signal.SIGINT, quit_handler)

    # Create payloadsdk object
    my_payload = PayloadSdkInterface()

    # Init payload
    my_payload.sdkInitConnection()
    print("Waiting for payload signal!\n")

    # Check connection
    my_payload.checkPayloadConnection()
    time.sleep(0.1)

    # Set gimbal RC mode to STANDARD 
    print("Set gimbal RC mode")
    my_payload.setPayloadCameraParam(PAYLOAD_CAMERA_RC_MODE, payload_camera_rc_mode.PAYLOAD_CAMERA_RC_MODE_STANDARD, param_type.PARAM_TYPE_UINT32)
    time.sleep(0.1)  

    # Move gimbal yaw to 90 deg
    print("Move gimbal yaw to 90 deg, delay in 5secs")
    my_payload.setGimbalSpeed(0, 0, 90, input_mode_t.INPUT_ANGLE)
    time.sleep(5)  

    # Move gimbal yaw to -90 deg
    print("Move gimbal yaw to -90 deg, delay in 5secs")
    my_payload.setGimbalSpeed(0, 0, -90, input_mode_t.INPUT_ANGLE)
    time.sleep(5)  

    # Move gimbal yaw to 0 deg
    print("Move gimbal yaw to 0 deg, delay in 5secs")
    my_payload.setGimbalSpeed(0, 0, 0, input_mode_t.INPUT_ANGLE)
    time.sleep(0.5) 

    # Close payload interface
    try:
        my_payload.sdkQuit()
    except Exception as e:
        print(f"Error while quitting payload: {e}")

if __name__ == "__main__":
    main()